Dockerfile

```diff
@@ -1,24 +1,28 @@
-FROM node:14.18.0-alpine as build
+FROM node:16-alpine as build
 
 ARG COMMIT_SHA=<not-specified>
-ENV NODE_ENV=production
 
 WORKDIR /build-dir
 
 COPY package.json .
 COPY package-lock.json .
 
-RUN npm ci
+# install also DEV dependencies to enable Typescript compilation
+RUN npm ci --production=false
 
 COPY . .
 
+# compile Typescript to Javascript
 RUN npm run build
 
+# remove unneeded files
+RUN rm -rf node_modules src tsconfig.json .swcrc
```

Comment:
you don't need this, this is a multi-stage build and these layers are automatically discarded